pandas/tools/tests/test_util.py

```diff
@@ -291,6 +291,45 @@ def test_non_hashable(self):
         with self.assertRaisesRegexp(TypeError, "Invalid object type"):
             pd.to_numeric(s)
 
+    def test_compact_ints(self):
+        s = ['1', 2, 3]
+
+        res = pd.to_numeric(s)
+        expected = np.array([1, 2, 3], dtype=np.int64)
+        tm.assert_numpy_array_equal(res, expected)
+
+        res = pd.to_numeric(s, compact_ints=True)
+        expected = np.array([1, 2, 3], dtype=np.int8)
+        tm.assert_numpy_array_equal(res, expected)
+
+        res = pd.to_numeric(s, compact_ints=True, use_unsigned=True)
+        expected = np.array([1, 2, 3], dtype=np.uint8)
+        tm.assert_numpy_array_equal(res, expected)
+
+        # ignore use_unsigned (negative number)
+        s = ['-1', 2, 3]
+        res = pd.to_numeric(s, compact_ints=True, use_unsigned=True)
+        expected = np.array([-1, 2, 3], dtype=np.int8)
+        tm.assert_numpy_array_equal(res, expected)
+
+        # ignore compact_ints (float element)
+        s = ['-1.0', 2, 3]
+        res = pd.to_numeric(s, compact_ints=True)
+        expected = np.array([-1, 2, 3], dtype=np.float64)
+        tm.assert_numpy_array_equal(res, expected)
+
+        # ensure behaviour is respected when values are int
+        s = [-1, 2, 3]
+        res = pd.to_numeric(s, compact_ints=True)
+        expected = np.array([-1, 2, 3], dtype=np.int8)
+        tm.assert_numpy_array_equal(res, expected)
```

Comment:
Can you explicitly add a check for `pd.to_numeric(np.array(s), compact_ints=True)` and ``pd.to_numeric(Series(s), compact_ints=True)`` as well? As the list case did already work before your changes (because those were converted to an object array, contrasting to an integer array when passing array/series)